### DISCUSSION TITLE
Use the data object with Alerts

### DISCUSSION THREAD

**Discussion by IAmMrTh** [gapitio/gapit-htmlgraphics-panel#119] [Q&A] [answered]
Hey,

I am using the Plugin (V.2.1.1) with Grafana (V.8.5.1). 
I am trying to get the status of Alerts, which are defined in Prometheus, with the data object. I am using the following PromQL expression: 
`ALERTS{instance="<IP>", alertstate="firing" `
The problem is that I only get 1 as output. Somehow it makes sence because ALERTS does not give an output when the Alert is not firing. So the last ouput is always 1. But do you maybe have a solution or a workaround so you can see if an alarm is firing or not?

**Comment by ZuperZee** [comment, marked as answer]
You can check the label of the query `data.series[X].fields[1].labels.alertstate`
![image](https://user-images.githubusercontent.com/39485579/186393453-aa9528bc-1ec5-4491-9d1f-7c87675e23cf.png)

**Reply by IAmMrTh** [reply]
Thanks for the fast answer!
The problem is the alerstate label does not change also when the alert is not firing. Does it change on your setup?

**Reply by ZuperZee** [reply]
Looks like multiple series are created depending on the alertstate. First pending will show, then a firing will show as a different series (if they're in the same time range). Both will have the same series name.
It doesn't look like a new series is made if the alert is ok (not firing or pending).
If you know the interval of your alert, you could check if the time is more than the last series time from the time field.
![image](https://user-images.githubusercontent.com/39485579/186425795-b3f0023d-66ab-499d-b68a-5ffcbf84e19a.png)

If the alert is outside the time range the firing and pending series are removed.

A simple example:
![image](https://user-images.githubusercontent.com/39485579/186431149-d8e65aab-c684-4e37-9187-77485906e852.png)

`Panel options`
```json
{
    "calcsMutation": "standard",
    "reduceOptions": {
        "calcs": [
            "lastNotNull",
            "last",
            "firstNotNull",
            "first",
            "min",
            "max",
            "mean",
            "sum",
            "count",
            "range",
            "delta",
            "step",
            "diff",
            "logmin",
            "allIsZero",
            "allIsNull",
            "diffperc"
        ]
    },
    "add100Percentage": true,
    "centerAlignContent": true,
    "overflow": "visible",
    "useGrafanaScrollbar": true,
    "SVGBaseFix": true,
    "codeData": "{\n  \"text\": \"Random text\"\n}",
    "rootCSS": "",
    "css": "* {\n  font-family: Open Sans;\n}\n",
    "html": "<div>\r\n</div>",
    "renderOnMount": true,
    "onRender": "const MY_ALERT_SERIES_NAME = \"My alert series\"\r\n\r\nconst alertSeriesList = htmlGraphics.data.series.filter(series => series.refId === MY_ALERT_SERIES_NAME);\r\n\r\nconst rootElt = htmlGraphics.htmlNode.querySelector(\"div\");\r\nrootElt.innerHTML = \"\"; // Clear children\r\n\r\nfor (const alertSeries of alertSeriesList) {\r\n  const [timeField, valueField] = alertSeries.fields;\r\n  const { alertstate, instance } = valueField.labels;\r\n  const epochAlertTime = timeField.values.get(timeField.values.length - 1);\r\n  const lastAlertTimeInMinutes = Math.floor((Date.now() - epochAlertTime) / 1000 / 60);\r\n\r\n  const newDivElt = document.createElement(\"div\");\r\n  newDivElt.textContent = `${instance} - ${alertstate} - ${lastAlertTimeInMinutes} minutes ago`\r\n  rootElt.append(newDivElt);\r\n}\r\n",
    "dynamicHtmlGraphics": false,
    "dynamicData": false,
    "dynamicFieldDisplayValues": false,
    "dynamicProps": false,
    "panelupdateOnMount": true,
    "onInitOnResize": false,
    "onInit": ""
}
```

**Reply by IAmMrTh** [reply]
Nice! That works! I did not know that i can see the timestamp from the last data object. But makes sence when there is a object called time :D
Thanks alot for the help!
